clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/DatabaseHandlerTest.java

```diff
@@ -104,4 +133,29 @@ public void empty_bundle_is_returned_if_no_db_connection() throws Exception {
         assertEquals(ClusterStateBundle.empty(), retrievedBundle);
     }
 
+    @Test
+    public void save_wanted_state_sheds_orphaned_states() throws Exception {
```

Comment:
Consider changing test name to something like `save_wanted_states_preserves_existing_wanted_state_mappings` to more accurately reflect its semantics

Reply:
Done